src/main/java/com/zepben/evolve/services/common/BaseService.kt

```diff
@@ -44,6 +47,8 @@ abstract class BaseService(
     val supportedClasses: Set<Class<out IdentifiedObject>> = Collections.unmodifiableSet(addFunctions.keys.map { it.java }.toSet())
     val supportedKClasses: Set<KClass<out IdentifiedObject>> get() = addFunctions.keys
 
+    private var _nameTypes: MutableMap<String, NameType>? = null
```

Comment:
Happy for this map to be non null and always exist. You don't have lots of service instances and it makes the code easier to read/write when not having to deal with it being nullable and mutable.